lib/cisco_node_utils/yum.rb

```diff
@@ -32,9 +33,16 @@ def self.decompose_name(file_name)
       # ex: b+z-ip2.x64_64
       name_arch_regex = /^([\w\-\+]+)\.(\w+)$/
 
-      file_name.match(name_ver_arch_regex) ||
-        file_name.match(name_ver_arch_regex_nx) ||
-        file_name.match(name_arch_regex)
+      # xrv9k-k9sec-1.0.0.0-r61102I.x86_64.rpm-XR-DEV-16.02.22C
+      name_arch_regex_xr = /^(.*\d.*)-([\d.]*)-(r\d\d\d\d*.).(\w{4,}).rpm/
```

Comment:
The above aren't super clear, but this seems to me like it should be `name_ver_arch_regex_xr` as it's parallel to `name_ver_arch_regex_nx`, not `name_arch_regex`.